src/snowflake/snowpark/modin/plugin/extensions/series_overrides.py

```diff
@@ -203,9 +207,22 @@ def item(self):  # noqa: RT01, D200
     pass  # pragma: no cover
 
 
-@register_series_not_implemented()
-def items(self):  # noqa: RT01, D200
-    pass  # pragma: no cover
+# Snowpark pandas has a custom iterator.
+@register_series_accessor("items")
+def items(self):
+    # TODO: SNOW-1063347: Modin upgrade - modin.pandas.Series functions
+    """
+    Iterate over ``Series`` rows as (index, value) tuples.
+    """
+
+    def items_builder(s):
+        """Return tuple of the given ``Series`` in the form (index, value)."""
+        return s.name, s.squeeze()
+
+    # Raise warning message since Series.items is very inefficient.
+    WarningMessage.single_warning(SERIES_ITEMS_WARNING_MESSAGE.format("Series.items"))
+
+    return SnowparkPandasRowPartitionIterator(DataFrame(self), items_builder, True)
```

Comment:
There was some controversy about the `enable_partition_with_native_pandas=True` option when it was introduced and I think it was originally set to `False`. I don't have time to investigate right now. looks like iterrows and itertuples do different things.

Reply:
I followed what is done in `itertuples()`. Experimenting around with it, significantly more queries are issued when the flag is set to False so I think this is reasonable to partition with pandas and throw the warning here.
```
    enable_partition_with_native_pandas: bool, default False
        When True, retrieve the table as partitions. Each partition is a pandas DataFrame which is iterated over until
        exhausted, and the next partition is pulled in.
        When False, iterate over the Snowpark pandas DataFrame directly row-by-row.
```

Reply:
`enable_partition_with_native_pandas=True` is likely to be faster for common use cases, but I don't have confidence in its implementation because we [skip](https://github.com/snowflakedb/snowpark-python/blame/560d56318208d9285c381897a46f0e747755820c/tests/integ/modin/frame/test_itertuples.py#L156) all the test cases that do `itertuples()` on large dataframes.

For `items()`, can we test `PARTITION_SIZE`, `PARTITION_SIZE - 1`, `PARTITION_SIZE + 1`, `PARTITION_SIZE * 2`, and `PARTITION_SIZE * 2 + 1`, and check manually that the those test cases are not too slow?

Reply:
Added a test, I experimented with the parameter set to True and it takes around 5-6 for `PARTITION_SIZE` and about 13 seconds for `PARTITION_SIZE * 2 + 1`. Set to False is takes significantly longer in both cases so I think having the parameter at True here is okay especially given the warning we provide.